Regular-Question-Answers/is-subsequence.cpp

```diff
@@ -0,0 +1,40 @@
+// Author: Bilgenur Çelik
+// Reviewer: Novruz Amirov
+// Question Link: https://leetcode.com/problems/is-subsequence/
+
+class Solution {
+public:
+    /* We want to know if the t string contains the s string's letters as a subsequence. 
+    *  With two pointers one pointing at the beginning of the t, another one to the s.
+    *  Move along the t string's letters. If the letter matches the s pointer's letter, point to the next value.
+    *  If the s pointer's value is equal to the length of the s, it is a subsequence.
+    *  If yes return true, else false.
+    */
+    bool isSubsequence(string s, string t) {
+        // base case:
+        if (s == "")
+            return true;
+
+        // sPointer will increase if letters match.
+        int sPointer = 0;
+
+        // tPointer is the t's pointer. It will increase whether or not the characters match.
+        for (int tPointer = 0; tPointer < t.length(); tPointer++){
+            // into a matching state:
+            if (t[tPointer] == s[sPointer]){
+                // increase sPointer
+                sPointer++;
+
+                // if the s array comes to an end, then it is a subsequence of t
+                if (sPointer == s.length())
+                    return true;
+            }
+        }
+
+        // in any other occasion:
+        return false;
+    }
+};
+
+// Time comp. -> O(n) Space comp. -> O(1)
+

```

Comment:
Good job! but it will be better if you explain what is n in O(n), something like "n -> then length of the string t"

Reply:
And delete the empty line 40